pages/admin/dashboard.js

```diff
@@ -0,0 +1,229 @@
+import { useRouter } from "next/router";
+import { useEffect, useState } from "react";
+import { CircularProgress, Typography } from "@material-ui/core";
+import PropTypes, { array } from "prop-types";
+import styles from "../../styles/GameRow.module.css";
+import game from "../../helpers/game";
+
+import firebase from "../../utils/firebase";
+import AdminGameView from "./game";
+
+import Link from "next/link";
+import {
+  Box,
+  Button,
+  Card,
+  CardActions,
+  CardContent,
+  CardHeader,
+  Container,
+  IconButton,
+  Paper,
+  Snackbar,
+  TextField,
+} from "@material-ui/core";
+
+export default function Dashboard() {
+  const [gameArrayState, setGameArrayState] = useState([]);
+  const [gameClick, setGameClick] = useState(false);
+  const [login, setLogin] = useState(false);
+  const [gamePin, setGamePin] = useState();
+
+  const router = useRouter();
+
+  if (firebase.apps.length === 0) {
+    console.log("not initialized");
+  } else {
+    firebase.auth().onAuthStateChanged(function (user) {
+      if (user) {
+        setLogin(true);
+      } else {
+        //no user logged in, so go back to admin
+        setLogin(false);
+        router.replace(`/admin?login=signOut`);
+      }
+    });
+  }
```

Comment:
Drop this in a `useEffect()` like the function below. Then get rid of the if statement.

Because of the nature of Next.js, it'll try to make every static. So you need to put it in a `useEffect()` to explicitly tell it to run it client side.